superset/views/geocoding.py

```diff
@@ -230,200 +223,174 @@ def geocode(self) -> Response:
+        :param if_exists: What should happen if the columns already exist
+        :param lat_column: The name of the latitude column
+        :param lon_column: The name of the longitude column
+        :param table_id: The id of the table
+        :return: A message describing what columns have been created or an empty message if no columns have been created
+        """
+
+        connection = table.database.get_sqla_engine().connect()
+        transaction = connection.begin()
+        try:
+            if not lat_exists:
+                self._add_column(connection, table, lat_column, Float())
+            if not lon_exists:
+                self._add_column(connection, table, lon_column, Float())
+        except Exception as e:
+            transaction.rollback()
+            raise SqlAddColumnException(
+                "An error occured while creating new columns for latitude and longitude",
+                e,
             )
+        finally:
+            transaction.close()
+            connection.close()
+            db.session.commit()
 
-        return column_message_suffix
+    def _get_table_with_columns(self, table_id: str) -> SqlaTable:
+        """
+        Check if there is a table and the table has columns
+        :param table_id: The id of the table
+        :raise TableNotFoundException: If there is no table object
+        :raise NoColumnsException: If the table does not have any columns
+        """
+        table = db.session.query(SqlaTable).filter_by(id=table_id).first()
+        if not table:
+            raise TableNotFoundException(f"Table with ID {table_id} does not exists")
+        if not table.columns:
+            raise NoColumnsException(f"No columns found for table with ID {table.id}.")
+        return table
 
     def _does_column_name_exist(self, table: SqlaTable, column_name: str):
         """
         Check if column name already exists in table
-        :param table_name: The table name of table to check
+        :param table: The SqlaTable
         :param column_name: The name of column to check
         :return true if column name exists in table
-        :raise TableNotFoundException: When there is no table
         """
-        if table and table.columns:
-            column_names = [column.column_name.lower() for column in table.columns]
-            return column_name.lower() in column_names
-        raise TableNotFoundException(f"Table with ID {table.id} does not exists")
+        column_names = [column.column_name.lower() for column in table.columns]
+        return column_name.lower() in column_names
 
     def _load_data_from_columns(
-        self, id: int, columns: list, lat_column=None, lon_column=None
-    ):
+        self, table: SqlaTable, columns: list
+    ) -> list:
         """
         Get data from columns form table
-        :param table_name: The table name from table from which select
+        :param table_id: The id of the table
```

Comment:
signature changes from id to table object but docstring changes from table-object to table id?